### PR TITLE
added blocks to respond with

### PR DESCRIPTION
**Pull request by elorest**
### Description of the Change

Adds blocks to `respond_with` as requested in https://github.com/amberframework/amber/issues/1053.

### Benefits

User code is easier to read.

### Possible Drawbacks

A bit more code.

### CODE REVIEW

**Comment by elorest** [comment]
Tests **ALL** pass locally. Not sure why're they failing on travis. Especially since the failing ones don't seem to have anything to do with my changes.

File: src/amber/controller/helpers/responders.cr [hunk 1]
```diff
@@ -24,32 +24,51 @@ module Amber::Controller::Helpers
       end
 
       def js(js : String | ProcType)
```
Comment (Sija) [inline src/amber/controller/helpers/responders.cr]: I'd suggest refactoring this with macros:

```cr
      {% for type in %w(html xml js json text) %}
        def {{type.id}}(value : String | ProcType)
          @available_responses[TYPE[:{{type.id}}]] = value
          self
        end

        def {{type.id}}(&block : -> _)
          {{type.id}}(block)
        end
      {% end %}

      def json(value : Hash(Symbol | String, String))
        json(value.to_json)
      end

      def json(**args : Object)
        json(args.to_h)
      end
```

Comment (elorest) [reply src/amber/controller/helpers/responders.cr]: Less readable but definitely a good idea. I'll do that. 

While your here any idea why the specs won't pass in Travis. They work perfectly on all my local and work computers.

Comment (Sija) [reply src/amber/controller/helpers/responders.cr]: I'm seeing `Error: formatting './src/amber/environment/settings.cr' produced changes`, could it be this?

Comment (Sija) [reply src/amber/controller/helpers/responders.cr]: After this refactor `add_response` method becomes obsolete.

Comment (elorest) [reply src/amber/controller/helpers/responders.cr]: Thank you. Interestingly Travis passed at the time. I guess format works differently in 0.27.1

I was still running format and specs with 0.27.0

**Comment by elorest** [comment]
Unfortunately travis still doesn't pass. Still is just fine locally. :(

**Comment by Sija** [comment]
@elorest Now for a change _Crecto_ specs are failing.

**Comment by elorest** [comment]
@sija yeah I noticed that. Can't make them fail in osx or arch though locally. I'm using 0.27.1 now so it should be the same.

**Comment by Sija** [comment]
@elorest That's odd... Do you run specs with Crecto test suite (same command as in Travis)?

**Comment by elorest** [comment]
@Sija Yep. `crystal spec ./spec/build_spec_crecto.cr`

File: src/amber/controller/helpers/responders.cr (hunk 1 above)
Comment (elorest) [reply src/amber/controller/helpers/responders.cr]: I've tried getting it to work with 0.27.2 now as well with no luck. Works locally but not on travis.

**Comment by elorest** [comment]
I've tried getting it to work with 0.27.2 now as well with no luck. Works locally but not on travis.

**Comment by Sija** [comment]
@elorest Does Crecto itself works with Crystal 0.27.1+?

**Pull request merged**